model/gen-ai/spans.yaml

```diff
@@ -71,6 +71,19 @@ groups:
         requirement_level: recommended
       - ref: gen_ai.usage.output_tokens
         requirement_level: recommended
+      - ref: gen_ai.conversation.id
+        requirement_level:
+          conditionally_required: when available
+        note: >
+          It helps maintain conversational state across multiple API calls. By consistently using the same gen_ai.conversation.id,
+          the system can:
+          1) Track the dialogue history and context across turns.
+          2) Improve coherence and continuity in responses.
+          3) Enable context-aware tool usage or memory mechanisms (e.g., referring back to earlier user inputs).
+          4) Facilitate multi-turn workflows such as task completion, follow-ups, or corrections.
+
+          Always generate a new unique gen_ai.conversation.id when initiating a new, unrelated conversation. Reuse the existing ID
+          only when continuing an ongoing interaction thread.
```

Comment:
I agree with these points, but they are targeted for application authors, not for client lib instrumentations, right? Instrumentation should no generate conversation id and cannot manage chat history. 

I think @drewby meant that it would be useful to elaborate when instrumentation should populate it, not when application should use it.

Reply:
updated.